docs/reference/configuration.txt

```diff
@@ -131,100 +131,84 @@ for details on driver options.
+          #   w: 1
 
           # Change the default read preference. Valid options for mode are: :secondary,
           # :secondary_preferred, :primary, :primary_preferred, :nearest
           # (default: primary)
-          read:
-            mode: :secondary_preferred
-            tag_sets:
-              - use: web
+          # read:
+          #   mode: :secondary_preferred
+          #   tag_sets:
+          #     - use: web
 
           # The name of the user for authentication.
-          user: 'user'
+          # user: 'user'
 
           # The password of the user for authentication.
-          password: 'password'
+          # password: 'password'
 
           # The user's database roles.
-          roles:
-            - 'dbOwner'
-
-          # Change the default authentication mechanism. Please see the
-          # driver documentation linked above for details on how to configure
-          # authentication. Valid options are :aws, :gssapi, :mongodb_cr,
-          # :mongodb_x509, :plain, :scram and :scram256 (default on 3.0
-          # and higher servers is :scram, default on 2.6 servers is :plain)
-          auth_mech: :scram
-
-          # Specify the auth source, i.e. the database or other source which
-          # contains the user's login credentials. Allowed values for auth source
-          # depend on the authentication mechanism, as explained in the server documentation:
-          # https://mongodb.com/docs/manual/reference/connection-string/#mongodb-urioption-urioption.authSource
-          # If no auth source is specified, the default auth source as
-          # determined by the driver will be used. Please refer to:
-          # https://mongodb.com/docs/ruby-driver/current/reference/authentication/#auth-source
-          auth_source: admin
-
-          # Connect directly to and perform all operations on the specified
-          # server, bypassing replica set node discovery and monitoring.
-          # Exactly one host address must be specified. (default: false)
-          #direct_connection: true
-
-          # Deprecated. Force the driver to connect in a specific way instead
-          # of automatically discovering the deployment type and connecting
-          # accordingly. To connect directly to a replica set node bypassing
-          # node discovery and monitoring, use direct_connection: true instead
-          # of this option. Possible values: :direct, :replica_set, :sharded.
-          # (default: none)
-          #connect: :direct
-
-          # Change the default time in seconds the server monitors refresh their status
+          # roles:
+          #   - 'dbOwner'
+
+          # Change the default authentication mechanism. Valid options are: :scram,
+          # :mongodb_cr, :mongodb_x509, and :plain. Note that all authentication
+          # mechanisms require username and password, with the exception of :mongodb_x509.
+          # Default on mongoDB 3.0 is :scram, default on 2.4 and 2.6 is :plain.
```

Comment:
MongoDB 2.4/2.6 should not be supported by latest Mongoid. Anyone using them at this point is gonna have a bad time...

Reply:
Maybe something like:
```
# Default is :scram (since MongoDB 3.0). Note the previous default for MongoDB 2.4 and 2.6 was :plain.
```